test/jasmine/tests/gl_plot_interact_basic_test.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+var Plotly = require('@lib/index');
+var Plots = require('@src/plots/plots');
+
+var createGraphDiv = require('../assets/create_graph_div');
+var destroyGraphDiv = require('../assets/destroy_graph_div');
+var mouseEvent = require('../assets/mouse_event');
+
+function teardown(gd, done) {
+
+    // The teardown function needs information of what to tear down so afterEach can not be used without global vars.
+    // In addition to removing the plot from the DOM it also destroy possibly present 2D or 3D scenes
+
+    // TODO we should figure out something to only rely on public API calls
```

Comment:
@etpinard if you have a suggestion for using a public API call for the destroy procedure, I'm glad to switch to it. In any case, general feedback is useful as it's a slightly different way of teardown compared to the global var + afterEach method. The purpose is to be as local about state management as possible, i.e. fewer conventions such as the gd global.